vm/integer.go

```diff
@@ -383,6 +383,21 @@ func builtinIntegerInstanceMethods() []*BuiltinMethodObject {
 				}
 			},
 		},
+		// Returns the `Decimal` conversion of self.
+		//
+		// ```Ruby
+		// 100.to_d # => '100'.to_d
+		// ```
+		// @return [Decimal]
+		{
+			Name: "to_d",
+			Fn: func(receiver Object, sourceLine int) builtinMethodBody {
+				return func(t *thread, args []Object, blockFrame *normalCallFrame) Object {
+					r := receiver.(*IntegerObject)
```

Comment:
Can you add argument check here?

Reply:
I'll do that. 

I also found that arg checks are missing in other methods so I'd add them in another PR.